arch/arm/soc/ti_simplelink/cc32xx/soc.h

```diff
@@ -21,11 +21,11 @@ enum {
 	Reset_IRQn                    = -15,
 	NonMaskableInt_IRQn           = -14,
 	HardFault_IRQn                = -13,
-#if defined(CONFIG_ARMV7_M)
+#if defined(CONFIG_ARMV7_M) || defined(CONFIG_ARMV8_M_MAINLINE)
```

Comment:
I'm not seeing why this addition is needed, as the cc32xx soc is an ARMV7_M.

Under what conditions would CONFIG_ARMV8_M_MAINLINE be defined for the cc32xx?

It seems programs including this cc32xx soc.h file would have SOC_SERIES_CC32XX defined, which through arch/arm/sock/ti_simplelink/cc32xx/Kconfig.series, selects CPU_CORTEX_M4, which via arch/arm/core/cortex_m/Kconfig then selects ARMV7_M.

Reply:
Any idea why the ifdef is there? Or any issue if we remove it?

Reply:
This is taken originally from cmsis.h.
But, by the reasoning above (i.e., that CONFIG_ARMV7_M must be set anyways), there should be no issue to remove it.

So, I just removed it from soc.h, and sanitycheck passed.  Result: PR#6009.

So, @ioannisg, I think you can remove the change to ti_simplelink/cc32xx/soc.h from your commit.

Reply:
Removed.